### DISCUSSION TITLE
Battery indicator

### DISCUSSION THREAD

**Discussion by TheTimC** [markqvist/Reticulum#492] [Ideas]
Hi mark et al,

I have successfully set up my first rnode using a heltec v2 board and 3d printed case. I've configured it with bluetooth, and I'm eagerly waiting for bluetooth to work on the v3 boards as I have them too.

One thing I've noticed is that there is no battery indicator on the display. It just behaves as if it is plugged into USB power while the battery is connected. It would be great if the display could determine if it is plugged in and charging, vs running from the battery and have a charge indicator. Is this possible?

![photo_2024-04-23_12-41-16](https://github.com/markqvist/Reticulum/assets/165362704/14077b86-6da1-43dc-8d29-8f18d4c53eb3)

Many thanks,
Tim.

**Comment by cobraPA** [comment]
Nice build!  What are you looking to connect to Bluetooth on the Heltec V3?  I don't think anyone has looked at the BLE version for Android yet.  The Heltec V3 only has a BLE chipset, so won't work with the Android Sideband for a bit until/if someone starts looking at a BLE version there.  It will work with the iPhone port, but that is not setup to merge with Mark's version yet.

**Reply by markqvist** [reply]
Regarding BLE, I recently changed a few things around in the structure for BLE startup and init on the ESP32-S3 chips. I'm still really not clear on what the best approach is. It's possible to use one of the "non-official", vendor-specific UART GATTs. I believe this is your current approach, right @cobraPA? Or we could make a custom "RNode GATT" or something like that. I'm really not sure what the best approach is here, and we should probably move it to a new discussion topic, if anyone wants to take the lead on that :)

**Reply by cobraPA** [reply]
Hmm, ok, I had a look... Are you assuming that only one of Bluetooth or BLE will be enabled?  For the iOS version, for now I have split out Bluetooth.h and a BLE_Rnode file.  I had not used bluefruit as I think was the case with the RAK/SX1262 changes.  And was thinking for things like Heltec V2 that we might want to turn on Classic Bluetooth and BLE, but using different libraries.  This would require using BLEServer.h, BLEDevice.h and BLEUtils.h instead of bluefruit.   My (somewhat limited!) understanding is that the BLE* support are current Arduino libraries with bluefruit being Adafruits version?
To answer your question for iOS, is uses these BLE libraries for the BLE interface then a custom UART emulator (really just ring buffers using your FIFO_Buffer code broken out.  I need to do another pass to see if I can use your main rnode handling loop, as it is duplicated for now in the BLE variant.
Also, Bluetooth pairing is not the same for BLE, from what I can see.  So I think we need to make some new handlers for that.  For example the iOS side does not enforce single paired devices by default, so multi-pairing is perfectly fine for iOS.  I will need to think some about how to enforce a similar behavior to what happens on standard bluetooth RNodes, if that is desired.  (This probably should move to a new discussion :) )

**Comment by markqvist** [comment]
The battery thing is most likely just telling the `Boards.h` file which ADC pin the battery voltage is connected to, and making a few accommodations for actually reading it in `Power.h`.
